util/stream_dataset.py

```diff
@@ -298,8 +299,32 @@ def fill_missing(self):
                 missing = self.missing.get(deployment, {}).get(param, {})
                 self._insert_missing(dataset, param, missing)
 
+    @staticmethod
+    def is_missing_arg_optional(missing_arg_map, param):
+        """
+        Check if the missing args are specified as optional in the function_parameter_map
+        of the parameter.
+        :param missing_arg_map: map of missing arguments for the function contained in the parameter
+        :param param: the Parameter who's parameter_function_map specifies the required args
```

Comment:
Now that these arguments may be optional, it seems you should restate "the required args"